File: Backend/test_gemini.js
const https = require('https');
require('dotenv').config();
const GEMINI_API_KEY = process.env.GEMINI_API_KEY || 'AIzaSyD5QwUhjz_b1Wm65O9qPDdU-vYuCW0lS-4';
const GEMINI_API_URL = 'https://generativelanguage.googleapis.com/v1beta/models/gemini-2.5-flash:generateContent?key=' + GEMINI_API_KEY;

const query = 'Harassment';
const prompt = `Generate exactly 5 short blog article summaries related to "${query}" for a women safety app.
Return ONLY a valid JSON array. Each object in the array must have these exact keys:
"type" (string, must be either "blog" or "article"), "title" (string), "description" (string, max 100 characters), "category" (string), "date" (string, format like "21 Apr 2026"), "url" (string, a valid working URL to a real article related to the topic).
Do not include any markdown formatting like \`\`\`json or \`\`\`. Just return the raw JSON array.`;

const requestBody = JSON.stringify({
    contents: [{
        parts: [{ text: prompt }]
    }]
});

const req = https.request(GEMINI_API_URL, {
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(requestBody)
    }
}, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => {
        console.log('Status:', res.statusCode);
        console.log('Response:', data);
    });
});
req.write(requestBody);
req.end();

File: frontend/test_gemini.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final prompt = '''
Generate 4 helpful and empowering articles or blogs about "Online Abuse" for a women's safety app.
Return ONLY a valid JSON object with a "blogs" array. Do not return markdown formatting like ```json.
Each object in the array should have these exact string keys:
- "title": A catchy, relevant title.
- "description": A 2-sentence summary.
- "content": A comprehensive, empowering 3-paragraph article/blog post providing actionable advice and support. Use plain text.
- "type": Either "Blog" or "Article".
- "category": A short category name (e.g. "Safety", "Legal", "Mental Health").
- "date": A date string like "Oct 12, 2023".
''';

  try {
    final response = await http.post(
      Uri.parse('https://generativelanguage.googleapis.com/v1beta/models/gemini-2.5-flash:generateContent?key=AIzaSyD5QwUhjz_b1Wm65O9qPDdU-vYuCW0lS-4'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        "contents": [{"parts": [{"text": prompt}]}],
        "generationConfig": {"temperature": 0.7}
      }),
    );

    print('Status: \${response.statusCode}');
    if (response.statusCode == 200) {
      final data = jsonDecode(response.body);
      final text = data['candidates'][0]['content']['parts'][0]['text'] as String;
      print('Raw text: \$text');
      final cleanText = text.replaceAll('```json', '').replaceAll('```', '').trim();
      final result = jsonDecode(cleanText);
      print('Parsed successfully. Found \${result['blogs']?.length} blogs.');
      print('First blog content: \${result['blogs'][0]['content'] != null ? 'Present' : 'Null'}');
    } else {
      print(response.body);
    }
  } catch (e) {
    print('Error: \$e');
  }
}
